libBlueSky/Dialog.hpp

```diff
@@ -23,29 +22,97 @@
 
 namespace BlueSky {
 
+    class SizeGrip;
+
     /**
      * @brief A Dialog
      *
      * This class automatically gives the Heaven::primaryWindow() as parent to the dialog.
      *
      */
-    class HEAVEN_BLUESKY_API Dialog : public QDialog
+    class HEAVEN_BLUESKY_API Dialog
+            : public QDialog
     {
         Q_OBJECT
+
+        class WatchDog;
+
+        friend class SheetEventFilter;
+
     public:
         Dialog();
         ~Dialog();
 
+    public:
+        bool isResizerEnabled() const;
+        void setResizerEnabled(bool enabled);
+
+    public slots:
+        virtual void open();
+
     protected:
-        void showEvent(QShowEvent*);
+#ifndef Q_OS_MAC
+        virtual void moveEvent(QMoveEvent* ev);
```

Comment:
The correct syntax would be `void moveEvent(QMoveEvent* ev) override;`. One should never declare a method virtual twice - Strange things can happen if you remove the virtual in the base... :)